shell/src/main/java/alluxio/cli/docgen/ConfigurationDocValidator.java

```diff
@@ -0,0 +1,176 @@
+
+  private ConfigurationDocValidator() {
+  } // prevent instantiation
+
+  /**
+   * Validates configuration files.
+   *
+   * @return returns number of deltas
+   */
+  public static Integer validate() throws IOException {
+    Collection<? extends PropertyKey> defaultKeys = PropertyKey.defaultKeys();
+    defaultKeys.removeIf(PropertyKey::isHidden);
+
+    // Sort defaultKeys
+    List<PropertyKey> dfkeys = new ArrayList<>(defaultKeys);
+    Collections.sort(dfkeys);
+
+    Integer numDeltas = 0;
+    numDeltas += validateFiles(dfkeys, CSV_FILE_DIR, CSV_FILE_NAMES);
+    numDeltas += validateFiles(dfkeys, YML_FILE_DIR, YML_FILE_NAMES);
+
+    return numDeltas;
+  }
+
+  private static Integer validateFiles(List<PropertyKey> dfkeys,
+                                       String fileDir, List<String> fileNames)
+          throws IOException {
+
+    String filePath = PathUtils.concatPath(Configuration.getString(PropertyKey.HOME), fileDir);
+    Integer numDeltas = 0;
+    for (String fileName : fileNames) {
+      String key = fileName.substring(0, fileName.indexOf("configuration") - 1);
+
+      // based on key substring within each filename, create a regex to
+      // filter for specific property keys.
+      List<String> rPatterns = new ArrayList<String>();
+      switch (key) {
+        case "user":
+          rPatterns.add("^alluxio\\.(user|user\\..*)");
+          break;
+        case "master":
+          rPatterns.add("^alluxio\\.(master|master\\..*)");
+          break;
+        case "worker":
+          rPatterns.add("^alluxio\\.(worker|worker\\..*)");
+          break;
+        case "security":
+          rPatterns.add("^alluxio\\.(security|security\\..*)");
+          break;
+        case "cluster-management":
+          rPatterns.add("^alluxio\\.(integration|integration\\..*)");
+          break;
+        case "common":
+          rPatterns.add("^alluxio\\.(?!(user|master|worker|security|integration)\\b)\\w*[\\.]?.*");
+          // additional regex pattern to retain non-system configuration records.
+          rPatterns.add("^(fs|s3a)\\b\\..*");
+          break;
+        default:
+          // ignore and skip any non-relevant files.
+          LOG.warn("Ignoring unrecognized filename substring key: " + key);
+          continue;
+      }
+
+      List<Pattern> compiledRegex = new ArrayList<Pattern>();
+      for (String pattern : rPatterns) {
+        compiledRegex.add(Pattern.compile(pattern));
+      }
+
+      List<String> revised = new ArrayList<String>();
+      if (fileDir.equals(CSV_FILE_DIR)) {
+        revised.add(CSV_FILE_HEADER);
+      }
+      for (PropertyKey propertyKey : dfkeys) {
+        String pKey = propertyKey.toString();
+        Boolean matched = false;
+        for (Pattern pattern : compiledRegex) {
+          if (pattern.matcher(pKey).matches()) {
+            matched = true;
+          }
+        }
+        if (matched) {
+          String pKeyDesc;
+          if (fileDir.equals(CSV_FILE_DIR)) {
```

Comment:
if you can if/else the file types in this common function, why can't the existing functions also express the logic the same way?